include/warcpp/warcpp.hpp

```diff
@@ -149,23 +188,83 @@ bool read_warc_header(std::istream &in, Warc_Record &record) {
  * 5. done
  *
  */
-bool read_warc_record(std::istream &in, Warc_Record &record) {
-    record.version_.clear();
-    record.fields_.clear();
-    record.content_.clear();
-    if (not read_warc_header(in, record)) {
-        return false;
+[[nodiscard]] auto read_record(std::istream &in) -> Result
+{
+    Record record;
+    if (auto error = detail::read_version(in, record.version_); error) {
+        return Result(*error);
+    }
+    if (auto error = detail::read_fields(in, record.fields_); error) {
+        return Result(*error);
+    }
+    if (not record.valid()) {
+        return Result(Missing_Mandatory_Fields{});
+    }
+    if (record.content_length() > 0) {
+        std::size_t length = record.content_length();
+        record.content_.resize(length);
+        if (not in.read(&record.content_[0], length)) {
+            return Result(Incomplete_Record{});
+        }
+    }
+    in.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    in.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    return Result(record);
+}
+
+[[nodiscard]] auto read_subsequent_record(std::istream &in) -> Result
+{
+    Record record;
+    std::optional<Invalid_Version> error;
+    while (error = detail::read_version(in, record.version_)) {
```

Comment:
`` `     using the result of an assignment as a condition without parentheses
      [-Wparentheses]
```